core/trino-main/src/main/java/io/trino/operator/aggregation/OrderedWindowAccumulator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.operator.PagesIndex;
+import io.trino.operator.window.PagesWindowIndex;
+import io.trino.spi.PageBuilder;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.ValueBlock;
+import io.trino.spi.connector.SortOrder;
+import io.trino.spi.function.WindowAccumulator;
+import io.trino.spi.function.WindowIndex;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class OrderedWindowAccumulator
+        implements WindowAccumulator
+{
+    private final PagesIndex.Factory pagesIndexFactory;
+    private final PagesIndex pagesIndex;
+    private final List<Type> argumentTypes;
+    private final List<Integer> argumentChannels;
+    private final List<Integer> sortChannels;
+    private final List<SortOrder> sortOrders;
+
+    private WindowAccumulator delegate;
```

Comment:
can you move non-final fields below final ones?

Reply:
I want to keep those together as they are very related to each other. And having delegate before initialDelegate is more natural for ordering of assignments in constructor. So keeping as is.